portal-impl/src/com/liferay/portal/service/impl/GroupLocalServiceImpl.java

```diff
@@ -861,18 +861,16 @@ public Group deleteGroup(Group group) throws PortalException {
 					group.getGroupId(), RoleConstants.TYPE_SITE);
 			}
 			else {
-				if (!group.isStagingGroup() || group.isStagedRemotely()) {
```

Comment:
This clause that you are removing was explicitly added in February @topolik in 853008d1c377aaa8

Tomas, do you agree with removing it now?